sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/testing/TestDataflowPipelineRunnerTest.java

```diff
@@ -145,7 +146,52 @@ public void testRunBatchJobThatFails() throws Exception {
     when(mockRunner.run(any(Pipeline.class))).thenReturn(mockJob);
 
     TestDataflowPipelineRunner runner = (TestDataflowPipelineRunner) p.getRunner();
-    runner.run(p, mockRunner);
+    try {
+      runner.run(p, mockRunner);
+      fail("AssertionError expected");
```

Comment:
I don't think this works (`fail` will throw an `AssertionError`, which will be caught inside this try/catch) - put fail outside of the catch block and return inside the catch